### ISSUE TITLE
Incorrect items as crafting result

### ISSUE THREAD

**Issue by acb** [Earlh21/TaintedCainCrafter#4]
I just started using your tool and while it's mostly great I've run into an issue where it frequently misidentifies what the result of the craft is.  Most of the time it's correct but many times it gets the item wrong.  I have a few examples that I've written down so far -- the item on the left is the one that your crafter says should be crafted, but the item on the right is what is actually made in game (item numbers in parentheses):

Dirty Mind (576) -> Psy Fly (581)
Yo Listen (492) -> Acid Baby (491)
Flush (288) -> Punching Bag (280)
Flat Stone (540) -> Pointy Rib (544)
Poke Go (505) -> Potato Peeler (487)
Trisagion (533) -> 7 Seals (526)
Rosary (72) -> PHD (75)

I couldn't quite figure out a pattern as to which items are wrong or why.  They all get pretty close to what it actually should be but are off a little one way or the other, sometimes getting an earlier item and sometimes getting a later item.  What I did notice is that even in the case of the ones that are off by a signification number of item IDs (e.g. Trisagion -> 7 Seals, 7 item IDs) they were much closer together in https://github.com/Earlh21/TaintedCainCrafter/blob/master/TaintedCain/Resources/Data/itempools.xml, usually 1 or 2 entries away from each other in one of the item pools in there.

Maybe there's an off by one error or a rounding error somewhere?  It's very strange.

**Comment by Earlh21** [comment]
Does this happen per-item or per-recipe? i.e. does the same item always get misidentified as the same wrong item?

**Comment by acb** [comment]
I haven't been able to nail that down yet, I've only gone through a couple runs with this tool.  I'll keep taking notes though.

I have an in-progress run that has the Trisagion -> 7 Seals mismatch (as well as a couple of these others), once it wraps up I'll reload the seed and spawn in pickups to get the "Trisagion" recipe and see if it still happens, if it does that'd be a useful test case for you.  I'd try to debug a bit more but I'm a lowly web dev and don't even know how to run .net applications haha.

**Comment by acb** [comment]
@Earlh21 I have a solid reproduction case for you: seed `QB67 0L7D` on Tainted Cain on hard.  If you spawn in 2 eternal hearts, 2 regular hearts, 2 keys, 1 bomb, and 1 penny that gets you a recipe for Trisagion in your app.  However, if you put those all in the bag it makes 7 Seals.  If you swap a heart and a key for 2 pennies that will make Trisagion in the game, but your app has that recipe attached to Marbles.  With the spawn below there will be 2 recipes for Trisagion and both will actually make 7 Seals.  If you put in the recipe for 7 Seals it does actually make 7 seals.

Interestingly I was able to confirm that the bug is recipe dependent and not always just an item switch.  I don't have specific details but I had 2 different recipes for Dirty Mind, one of which was actually Dirty Mind in game and one of which was Psy Fly.  I also had 3 BFFs recipes -- 2 made BFFs and one made...something else.  I forgot to write it down.  I don't think the BFFs recipe was anything exotic though, if you spawn a handful of pickups you should be able to get a few and try them out.

For reference, here's what I spawned for my Trisagion testing:
```
spawn 5.10.4
spawn 5.10.4
spawn 5.10.1
spawn 5.10.1
spawn 5.20.1
spawn 5.20.1
spawn 5.20.1
spawn 5.30.1
spawn 5.30.1
spawn 5.40.1
spawn 5.40.1
```

Basically, the steps are: 
1. Launch the game on hard with Tainted Cain, seed `QB67 0L7D`
2. Run the spawns from above
3. Send them to the crafting tool
4. At this point, the tool will have some recipes:
* 2 recipes for Trisagion, both of which will actually craft 7 Seals
* 1 recipe for 7 Seals, which will also craft 7 Seals
* 1 recipe for Marbles, which will actually craft Trisagion

**Comment by acb** [comment]
@Earlh21 I may have found another clue.  A couple weeks ago when I was using your tool I noticed that it had recipes available to me to craft The Swarm, but I just realized that I didn't have it unlocked at the time.  I just unlocked it last night.

Maybe something's up with it detecting what items are available, so if one is missing it makes the recipe be off slightly?  I saw that there's code that checks achievements to figure that out (I presume) so maybe fixing this would be as simple as exposing an interface where we can check/uncheck what we have unlocked manually, or at least validate that what the app has is correct.

**Comment by SchlongWhisperer** [comment]
I had the same issue, I tinkered a little bit and found out that it's just the files inside the data folder are not up to date, replacing the "itempools.xml" and "items_metadata.xml" from the "Resources\Data" file directory with the files from your game folder (after you have extracted them with the resource extractor tool) seems to fix the issue as far as I can tell, I tried @acb seed and Trisagion crafts correctly

replacing the items.xml seems to break the program as it doesn't start anymore, but the original seems to work fine even if there are differences, you also need to remove the trinket lines from the "itempools.xml" and "items_metadata.xml" if you don't the program won't start

I am attaching the 2 files with the appropriate changes already made, they work fine for the v1.7.8a of Isaac
[files.zip](https://github.com/Earlh21/TaintedCainCrafter/files/8384567/files.zip)

**Comment by Earlh21** [comment]
Thank you, @SchlongWhisperer, those files seem to fix the problem. Can you try those files, @acb, and see if you still get any incorrect recipes?

**Comment by acb** [comment]
I just got a chance to check this out and yeah, it seems to have made a big difference.  I do still sometimes see recipes for items that I don't have unlocked yet but that's a lot easier to work around and will resolve itself once I have everything unlocked.  Every recipe I've seen so far has been correct other than the ones for locked items.
